File: src/AdventureWorks.Application/DataEngine/Production/ProductListPriceHistory/Commands/Queries/GetProductListPriceHistories/ProductListPriceHistoryLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Production;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;


namespace AdventureWorks.Application.DataEngine.Production.ProductListPriceHistory.Queries.GetProductListPriceHistories
{
    public partial class ProductListPriceHistoryLookupModel : IHaveCustomMapping
    {
        public int ProductID { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public decimal ListPrice { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Production.ProductListPriceHistory, ProductListPriceHistoryLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<ProductListPriceHistoryLookupModel, BaseProductListPriceHistory>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/production/product-list-price-history/product-list-price-history-lookup-model.ts

export interface IProductListPriceHistoryLookupModel {
  productID: number;
  startDate: Date | string;
  endDate: Date | string | null;
  listPrice: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class ProductListPriceHistoryLookupModel implements IProductListPriceHistoryLookupModel {
  productID: number;
  startDate: Date | string;
  endDate: Date | string | null;
  listPrice: number;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<ProductListPriceHistoryLookupModel>) {
    if (!!init) {
      Object.assign<ProductListPriceHistoryLookupModel, Partial<ProductListPriceHistoryLookupModel>>(this, init);
    }
  }
}
